compiler/src/iree/compiler/Dialect/Stream/Transforms/EncodeTensors.cpp

```diff
@@ -33,6 +34,17 @@ namespace IREE {
 namespace Stream {
 namespace {
 
+//===----------------------------------------------------------------------===//
+// Bitwidth utilities
+//===----------------------------------------------------------------------===//
+static int64_t getTypeBitwidth(Type type) {
+  if (isa<ComplexType>(type)) {
+    return 2 *
+           dyn_cast<ComplexType>(type).getElementType().getIntOrFloatBitWidth();
+  }
```

Comment:
isa and dyn_cast are both expensive and dyn_cast is just isa + cast - so since you've checked here with isa you can just cast instead. even better, though, is to do it all at once - this does isa + casts if a match and otherwise returns {} such that the if fails:
```suggestion
  if (auto complexType = type.dyn_cast<ComplexType>()) {
    return 2 * complexType.getElementType().getIntOrFloatBitWidth();
  }
```